spotinst/resource_spotinst_ocean_aks_np_test.go

```diff
@@ -0,0 +1,902 @@
+func TestAccSpotinstOceanAKSNP_Headrooms(t *testing.T) {
+	clusterName := "Terraform-Test-AKS-2-0-Cluster"
+	controllerClusterID := "terraform-aks-2-0-cluster"
+	resourceName := createOceanAKSNPResourceName(clusterName)
+
+	var cluster azure_np.Cluster
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t, "azure") },
+		Providers:    TestAccProviders,
+		CheckDestroy: testOceanAKSNPDestroy,
+
+		Steps: []resource.TestStep{
+			{
+				Config: createOceanAKSNPTerraform(&OceanAKSNPMetadata{
+					clusterName:         clusterName,
+					controllerClusterID: controllerClusterID,
+				}),
+				//ExpectNonEmptyPlan: true,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckOceanAKSNPExists(&cluster, resourceName),
+					testCheckOceanAKSNPAttributes(&cluster, clusterName),
+					resource.TestCheckResourceAttr(resourceName, "headrooms.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "headrooms.0.cpu_per_unit", "1024"),
+					resource.TestCheckResourceAttr(resourceName, "headrooms.0.memory_per_unit", "512"),
+					resource.TestCheckResourceAttr(resourceName, "headrooms.0.gpu_per_unit", "0"),
+					resource.TestCheckResourceAttr(resourceName, "headrooms.0.num_of_units", "2"),
+				),
+			},
+			{
+				Config: createOceanAKSNPTerraform(&OceanAKSNPMetadata{
+					clusterName:          clusterName,
+					controllerClusterID:  controllerClusterID,
+					headrooms:            testHeadroomsOceanAKSNPConfig_Update,
+					updateBaselineFields: true,
+				}),
+				//ExpectNonEmptyPlan: true,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "headrooms.#", "2"),
+					resource.TestCheckResourceAttr(resourceName, "headrooms.0.cpu_per_unit", "1024"),
+					resource.TestCheckResourceAttr(resourceName, "headrooms.0.memory_per_unit", "512"),
+					resource.TestCheckResourceAttr(resourceName, "headrooms.0.gpu_per_unit", "0"),
+					resource.TestCheckResourceAttr(resourceName, "headrooms.0.num_of_units", "2"),
+					resource.TestCheckResourceAttr(resourceName, "headrooms.1.cpu_per_unit", "2048"),
+					resource.TestCheckResourceAttr(resourceName, "headrooms.1.memory_per_unit", "1024"),
+					resource.TestCheckResourceAttr(resourceName, "headrooms.1.gpu_per_unit", "2"),
+					resource.TestCheckResourceAttr(resourceName, "headrooms.1.num_of_units", "4"),
+				),
+			},
+		},
+	})
+}
+
+const testHeadroomsOceanAKSNPConfig_Create = `
+  // --- autoscale --------------------------------------------------------
+  headrooms {
+    cpu_per_unit    = 1024
+    memory_per_unit = 512
+    gpu_per_unit    = 0
+    num_of_units    = 2
+  }
+// ----------------------------------------------------------------------
+
+`
+const testHeadroomsOceanAKSNPConfig_Update = `
+  // --- autoscale --------------------------------------------------------
+  headrooms {
+    cpu_per_unit    = 1024
+    memory_per_unit = 512
+    gpu_per_unit    = 0
+    num_of_units    = 2
+  }
+
```

Comment:
why update has 2 block of headrooms and also include emptyfields check

Reply:
added everwhere except health